datetime/datetime_test.go

```diff
@@ -565,6 +698,31 @@ func TestMPDecode(t *testing.T) {
 	}
 }
 
+func TestUnmarshalMsgpackInvalidLength(t *testing.T) {
+	var v Datetime
+
+	err := v.UnmarshalMsgpack([]byte{0x04})
+	if err == nil {
+		t.Fatalf("Unexpected success %v.", v)
+	}
+	if err.Error() != "Invalid data length: got 1, wanted 8 or 16" {
+		t.Fatalf("Unexpected error: %s", err.Error())
+	}
+}
+
+func TestUnmarshalMsgpackInvalidZone(t *testing.T) {
+	var v Datetime
+
+	buf, _ := hex.DecodeString("b016b9430000000000000000b400ee01")
```

Comment:
I think it's worth do add a comment about what's been encoded here.

Reply:
Thank you, added:
```patch
@@ -713,6 +716,10 @@ func TestUnmarshalMsgpackInvalidLength(t *testing.T) {
 func TestUnmarshalMsgpackInvalidZone(t *testing.T) {
        var v Datetime
 
+       // The original value from datetimeSample array:
+       // {time.RFC3339 + " MST",
+       //  "2006-01-02T15:04:00+03:00 MSK",
+       //  "d804b016b9430000000000000000b400ee00"}
```